### PR TITLE
-chapter_4

### PR DESCRIPTION
**Pull request by chuan97sun**
ready

### CODE REVIEW

**Comment by chuan97sun** [comment]
ready

File: Gemfile [hunk 1]
```diff
@@ -44,4 +45,9 @@ group :test do
...
 end
 
-gem "tzinfo-data", platforms: [:mingw, :mswin, :x64_mingw, :jruby]
\ No newline at end of file
+gem "tzinfo-data", platforms: [:mingw, :mswin, :x64_mingw, :jruby]
+
+group :development, :test do
+  gem "rubocop", "~> 0.74.0", require: false
+  gem "rubocop-rails", "~> 2.3.2", require: false
+end
```
Comment (chungpv-1008) [inline Gemfile]: please add end line for this file

File: app/assets/stylesheets/custom.scss [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+font-size: 1.7em;
+color: #fff;
+text-transform: uppercase;
+letter-spacing: -1px;
+padding-top: 9px;
+font-weight: bold;
+}
+#logo:hover {
+color: #fff;
+text-decoration: none;
```
Comment (chungpv-1008) [inline app/assets/stylesheets/custom.scss]: add indent 2 spaces for this file, move color variables to other class, then call back here. Add end line for this file

File: app/helpers/application_helper.rb [hunk 3]
```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
+  def full_title page_title = ""
+    base_title = "Ruby on Rails Tutorial Sample App"
+    page_title.empty? ? base_title : "#{page_title} | #{base_title}"
```
Comment (chungpv-1008) [inline app/helpers/application_helper.rb]:
```suggestion
    page_title.blank? ? base_title : "#{page_title} | #{base_title}"
```

Comment (chungpv-1008) [reply app/helpers/application_helper.rb]: you try to find out how 'blank' different 'empty'

**Comment by chuan97sun** [comment]
ready

File: app/assets/stylesheets/users.scss [hunk 4]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Users controller here.
```
Comment (manhnt-0960) [inline app/assets/stylesheets/users.scss]: delete this file, it's empty

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -0,0 +1,4 @@
+class UsersController < ApplicationController
+  def new
```
Comment (manhnt-0960) [inline app/controllers/users_controller.rb]:
```suggestion
  def new; end
```

File: app/helpers/application_helper.rb [hunk 6]
```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
+  def full_title page_title = ""
```
Comment (manhnt-0960) [inline app/helpers/application_helper.rb]:
```suggestion
  def full_title page_title
```

File: app/helpers/users_helper.rb [hunk 7]
```diff
@@ -0,0 +1,2 @@
+module UsersHelper
```
Comment (manhnt-0960) [inline app/helpers/users_helper.rb]: delete this file

**Comment by chuan97sun** [comment]
ready

**Review by chungpv-1008 — COMMENTED** [review]
approved

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**